From e5e00a118bbf5d248b67469928a873b3855da1f4 Mon Sep 17 00:00:00 2001
From: "Kothapeta, BikshapathiX" <bikshapathix.kothapeta@intel.com>
Date: Fri, 10 Feb 2023 12:44:43 +0530
Subject: [PATCH] INTERNAL: Duplicate fd in iris driver instead of reuse

Duplicate DRM file descriptor internally instead of reuse.

Tracked-On: OAM-105812
Signed-of-by: Shekhar Chauhan <shekhar.chauhan@intel.com>
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>

diff --git a/src/gallium/drivers/iris/iris_screen.c b/src/gallium/drivers/iris/iris_screen.c
index d079a80d895..4b34f00ca31 100644
--- a/src/gallium/drivers/iris/iris_screen.c
+++ b/src/gallium/drivers/iris/iris_screen.c
@@ -625,8 +625,7 @@ iris_screen_destroy(struct iris_screen *screen)
    u_transfer_helper_destroy(screen->base.transfer_helper);
    iris_bufmgr_unref(screen->bufmgr);
    disk_cache_destroy(screen->disk_cache);
-   // fd close is handled by owner module
-   //close(screen->winsys_fd);
+   close(screen->winsys_fd);
    ralloc_free(screen);
 }
 
-- 
2.39.1

